api/v1alpha1/inferencemodel_types.go

```diff
@@ -17,71 +17,82 @@ limitations under the License.
 	// Names can be reserved without implementing an actual model in the pool.
 	// This can be done by specifying a target model and setting the weight to zero,
 	// an error will be returned specifying that no valid target model is found.
-	Name string `json:"name,omitempty"`
+	ModelName string `json:"modelName,omitempty"`
 	// Optional
-	// LLM Services with an objective have higher priority than services without.
-	// IMPORTANT: By specifying an objective, this places the LLMService in a higher priority class than LLMServices without a defined priority class.
-	// In the face of resource-scarcity. Higher priority requests will be preserved, and lower priority class requests will be rejected.
-	Objective *Objective `json:"objective,omitempty"`
+	// Defines how important it is to serve the model compared to other models referencing the same pool.
+	Criticality *Criticality `json:"criticality,omitempty"`
 	// Optional.
 	// Allow multiple versions of a model for traffic splitting.
 	// If not specified, the target model name is defaulted to the modelName parameter.
 	// modelName is often in reference to a LoRA adapter.
 	TargetModels []TargetModel `json:"targetModels,omitempty"`
+	// Reference to the InferencePool that the model registers to. It must exist in the same namespace.
+	PoolRef LocalObjectReference `json:"poolRef,omitempty"`
 }
 
+// LocalObjectReference identifies an API object within the namespace of the
+// referrer.
+type LocalObjectReference struct {
+	// Group is the group of the referent.
+	Group string `json:"group,omitempty"`
+
+	// Kind is kind of the referent. For example "InferencePool".
+	Kind string `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	Name string `json:"name,omitempty"`
+}
+
+// Defines how important it is to serve the model compared to other models.
+type Criticality string
+
+const (
+	// Most important. Requests to this band will be shed last.
+	Critical Criticality = "Critical"
+	// More important than Sheddable, less important than Critical.
+	// Requests in this band will be shed before critical traffic.
+	Default Criticality = "Default"
+	// Least important. Requests to this band will be shed before all other bands.
+	Sheddable Criticality = "Sheddable"
+)
+
 // TargetModel represents a deployed model or a LoRA adapter. The
 // Name field is expected to match the name of the LoRA adapter
 // (or base model) as it is registered within the model server. Inference
 // Gateway assumes that the model exists on the model server and is the
 // responsibility of the user to validate a correct match. Should a model fail
 // to exist at request time, the error is processed by the Instance Gateway,
-// and then emitted on the appropriate LLMService object.
+// and then emitted on the appropriate InferenceModel object.
 type TargetModel struct {
 	// The name of the adapter as expected by the ModelServer.
 	Name string `json:"name,omitempty"`
```

Comment:
Similar comments elsewhere, but _everything_ in this API should have min and max length validation, and strings should also have an allowlist of characters with the idea that we can relax validation in the future if necessary.

Reply:
same, comment as above on modelName, I don't think there is a spec, and they can include characters like `@` and `;`

Reply:
Even if we start with a really broad set of allowed characters, I'd rather start with some kind of allowlist than allowing everything. At a minimum, for the sake of computing CEL complexity, every single field needs to have a max length.

Reply:
Agreed on the max length, I just don't know what a reasonable set of allowed characters we can start with here given that there is no spec for model names.

Reply:
Also I dont think model names have to be in English, or use the english alphabet